applications/script.js

```diff
@@ -252,27 +253,63 @@ async function renderApplicationCards(next, status, isInitialRender) {
   noApplicationFoundText.classList.add('hidden');
   changeLoaderVisibility({ hide: false });
   isDataLoading = true;
-  const data = await getApplications({
-    applicationStatus: status,
-    next,
-  });
-  isDataLoading = false;
-  changeLoaderVisibility({ hide: true });
-  const applications = data.applications;
-  nextLink = data.next;
-  if (isInitialRender) filterButton.classList.remove('hidden');
-  if (!applications.length)
-    return noApplicationFoundText.classList.remove('hidden');
-  applications.forEach((application) => {
-    const applicationCard = createApplicationCard({
-      application,
+
+  try {
+    const data = await getApplications({ applicationStatus: status, next });
+    isDataLoading = false;
+    changeLoaderVisibility({ hide: true });
+
+    const applications = data.applications;
+    nextLink = data.next;
+
+    if (isInitialRender) filterButton.classList.remove('hidden');
+    if (!applications.length)
+      return noApplicationFoundText.classList.remove('hidden');
+
+    applicationContainer.innerHTML = '';
+    applications.forEach((application) => {
+      console.log('applications', application);
+
+      const applicationCard = createApplicationCard({ application });
+      applicationContainer.appendChild(applicationCard);
     });
+  } catch (error) {
+    console.error('Error fetching applications:', error);
+    isDataLoading = false;
+    changeLoaderVisibility({ hide: true });
+    noApplicationFoundText.classList.remove('hidden');
+  }
+}
+
+async function renderApplicationById(id) {
+  noApplicationFoundText.classList.add('hidden');
+  changeLoaderVisibility({ hide: false });
+  isDataLoading = true;
+
+  try {
+    const data = await getApplicationById(id);
+
+    const application = data.application;
+
+    isDataLoading = false;
```

Comment:
don't we need to handle this flag in catch block

Reply:
fixed